crates/solidity/outputs/cargo/tests/src/doc_examples/simple_contract.rs

```diff
@@ -14,12 +14,13 @@ fn simple_contract() -> Result<()> {
 
     let parse_tree = parse_output.tree();
 
-    let children = if let Node::Rule(rule) = &parse_tree {
+    let rule = if let Node::Rule(rule) = parse_tree {
         assert_eq!(rule.kind, RuleKind::ContractDefinition);
-        &rule.children
+        rule
     } else {
         panic!("Unexpected parse_tree");
     };
```

Comment:
nit, since `rule` is not used anywhere else, maybe we can test `rule.reconstruct()` in the original `if let` expression?

```suggestion
    let children = if let Node::Rule(rule) = &parse_tree {
        assert_eq!(rule.kind, RuleKind::ContractDefinition);
        assert_eq!(rule.reconstruct(), "contract Foo {}");
        &rule.children
    } else {
        panic!("Unexpected parse_tree");
    };
```

Or otherwise, using the existing `let rule = parse_tree.as_rule().expect(...)` and testing `rule` fully in one place.

Reply:
Not so keen on `reconstruct` as a name. Maybe `source_text` or something. Am I reconstructing the node? I had to go and look at the source when I first saw that name to find out what it meant.

https://medium.com/@a.pouryousefi98/meaningful-name-with-javascript-examples-part-1-564141c6e399

Which originally comes from 'Intention Revealing Selectors' best practice in Smalltalk. I like code to read like a conversation, or a story.